components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -156,39 +156,44 @@ func (o *odometry) Reconfigure(ctx context.Context, deps resource.Dependencies,
 	// check if new motors have been added, or the existing motors have been changed, and update the motorPairs accorodingly
 	for i := range newConf.LeftMotors {
 		var motorLeft, motorRight motor.Motor
-		if i >= len(o.motors) || o.motors[i].left.Name().ShortName() != newConf.LeftMotors[i] {
-			motorLeft, err = motor.FromDependencies(deps, newConf.LeftMotors[i])
-			if err != nil {
-				return err
-			}
-			properties, err := motorLeft.Properties(ctx, nil)
-			if err != nil {
-				return err
-			}
-			if !properties.PositionReporting {
-				return motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
-			}
+		var changed, added bool
+
+		motorLeft, err = motor.FromDependencies(deps, newConf.LeftMotors[i])
+		if err != nil {
+			return err
+		}
+		properties, err := motorLeft.Properties(ctx, nil)
+		if err != nil {
+			return err
+		}
+		if !properties.PositionReporting {
+			return motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
 		}
 
-		if i >= len(o.motors) || o.motors[i].right.Name().ShortName() != newConf.RightMotors[i] {
-			motorRight, err = motor.FromDependencies(deps, newConf.RightMotors[i])
-			if err != nil {
-				return err
-			}
-			properties, err := motorRight.Properties(ctx, nil)
-			if err != nil {
-				return err
-			}
-			if !properties.PositionReporting {
-				return motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
-			}
+		motorRight, err = motor.FromDependencies(deps, newConf.RightMotors[i])
+		if err != nil {
+			return err
+		}
+		properties, err = motorRight.Properties(ctx, nil)
+		if err != nil {
+			return err
+		}
+		if !properties.PositionReporting {
+			return motor.NewPropertyUnsupportedError(properties, newConf.LeftMotors[i])
+		}
+
+		if i >= len(o.motors) {
+			added = true
+		} else if (o.motors[i].left.Name().ShortName() != newConf.LeftMotors[i]) ||
+			(o.motors[i].right.Name().ShortName() != newConf.RightMotors[i]) {
+			changed = true
```

Comment:
These variables are each created in 1-liners, and used in a single place. Why not just inline them like before?

```go
if i >= len(o.motors) {
    o.motors = append(...)
// This next line is the only one that needs to change from the original
} else if (o.motors[i].leftName().ShortName() != newConf.LeftMotors[i]) || ...) {
    o.motors[i].left = motorLeft
    o.motors[i].right = motorRight
}
// Neither `added` nor `changed` are ever mentioned.
```

Reply:
good point. I'll remove them